zipline/data/us_equity_minutes.py

```diff
@@ -49,10 +48,29 @@ def write_metadata(directory, first_trading_day):
         json.dump(metadata, fp)
 
 
+def _bcolz_minute_index(trading_days):
+    minutes = np.zeros(len(trading_days) * MINUTES_PER_DAY,
+                       dtype='datetime64[ns]')
+    market_opens = tradingcalendar.open_and_closes.market_open
+    mask = market_opens.index.slice_indexer(start=trading_days[0],
```

Comment:
Looking at where this is being called, I think this truncation logic is actually just unnecessary.  The trading days here are being generated from `days_in_range`, so I think we should just make it the responsibility of the caller to pass in the array of opens.  

I'd probably write this as:

```
def trading_minute_range(opens, minutes_per_day):
    """
    Construct an array of datetime64[ns] with the ``minutes_per_day`` minutes for each
    day in ``opens``.

    Parameters
    ----------
    opens : array[datetime64[ns]]
        An array of daily market opens.
    minutes_per_day : int
        The number of minutes to produce for each day.

    Returns
    -------
    minutes : array[datetime64[ns]]
        An array containing ``minutes_per_day`` sequential minutes for each day in ``opens``.
    """
    return (opens.values[:, None] + arange(minutes_per_day, dtype='timedelta64[m]'))\
        .reshape((len(opens) * minutes_per_day,))
```
I'd probably just require that this be passed an array of the opens and the number of minutes in each day.